tests/unit-test.el

```diff
@@ -197,57 +197,89 @@
       (expect (cljr--language-context-at-point)
               :to-equal '("cljc" nil))))
 
-  ;; Marked as pending per discussion in https://github.com/clojure-emacs/clj-refactor.el/issues/533
-  (xit "identifies a cljc file with a cljs context"
+  (it "identifies a cljc file with a cljs context"
     (cljr--with-clojure-temp-file "foo.cljc"
       (with-point-at "(ns foo) #?(:cljs (Math/log|))"
         (expect (cljr--language-context-at-point)
                 :to-equal '("cljc" "cljs")))))
 
-  (xit "identifies a cljc file with a clj context"
+  (it "identifies a cljc file with a clj context"
     (cljr--with-clojure-temp-file "foo.cljc"
       (with-point-at "(ns foo) #?(:clj (Math/log|))"
         (expect (cljr--language-context-at-point)
                 :to-equal '("cljc" "clj")))))
 
-  (xit "identifies a nested context with two branches present"
+  (it "identifies a nested context with two branches present"
     (cljr--with-clojure-temp-file "foo.cljc"
       (with-point-at "(ns foo) #?(:cljs (Math/log|) :clj 1)"
         (expect (cljr--language-context-at-point)
                 :to-equal '("cljc" "cljs")))))
 
-  (xit "identifies a nested context with an alternate branch proceeding"
+  (it "identifies a nested context with an alternate branch proceeding"
     (cljr--with-clojure-temp-file "foo.cljc"
       (with-point-at "(ns foo) #?(:clj 1 :cljs (Math/log|))"
         (expect (cljr--language-context-at-point)
                 :to-equal '("cljc" "cljs")))))
 
-  ;; FIXME: the following cases cause emacs to lock up in a loop as as
-  ;; `cljr--point-in-reader-conditional-branch-p' or
-  ;; `cljr--goto-reader-conditional' have infinite loops.
-  (xit "returns nil in an incomplete reader conditional"
+  (it "identifies a nested context when point is on symbol"
+    (cljr--with-clojure-temp-file "foo.cljc"
+      (with-point-at "(ns foo) #?(:clj 1 |:cljs 2)"
+        (expect (cljr--language-context-at-point)
+                :to-equal '("cljc" "cljs")))))
+
+  (it "identifies a nested context if point is between symbol and form"
+    (cljr--with-clojure-temp-file "foo.cljc"
+      (with-point-at "(ns foo) #?(:cljs |1)"
+        (expect (cljr--language-context-at-point)
+                :to-equal '("cljc" "cljs")))))
+
+  (it "returns nil in an incomplete reader conditional"
     (cljr--with-clojure-temp-file "foo.cljc"
       (with-point-at "(ns foo) #?(|)"
         (expect (cljr--language-context-at-point)
                 :to-equal '("cljc" nil)))))
 
-  (xit "returns :default context if specified"
+  (it "returns last context in an incomplete reader conditional"
```

Comment:
The first example was causing an unbounded loop, so I fixed it to stop if (point) reaches the end of the file. I think this was an edge case that only triggers if it's literally the last character in the file but regardless it's addressed. 3rd example worked fine.

However, the second example fails as `cljr--beginning-of-reader-conditional` fails to find the enclosing reader conditional, because `cljr--top-level-p` erroneously identifies the form as top level. I've added tests demonstrating this for `cljr--language-context-at-point`, `cljr--beginning-of-reader-conditional` and `cljr--top-level-p`. The result of an unparsable file is that we can't identify the context at point correctly (it identifies it as nil), but it does terminate, so that seems like an acceptable result given that the input is bad.